code/datums/components/crafting/recipes/recipes_weapon_and_ammo.dm

```diff
@@ -394,3 +398,60 @@
 	time = 5
 	category = CAT_WEAPONRY
 	subcategory = CAT_AMMO
+
+////////////////////
+// PARTS CRAFTING //
+////////////////////
+
+// BARRELS
+
+/datum/crafting_recipe/rifle_barrel
+	name = "Improvised Rifle Barrel"
+	result = /obj/item/weaponcrafting/improvised_parts/barrel_rifle
+	reqs = list(/obj/item/pipe = 2)
+	tools = list(TOOL_WELDER,TOOL_SAW)
+	time = 150
+	category = CAT_WEAPONRY
+	subcategory = CAT_PARTS
+
+/datum/crafting_recipe/shotgun_barrel
+	name = "Improvised Shotgun Barrel"
+	result = /obj/item/weaponcrafting/improvised_parts/barrel_shotgun
+	reqs = list(/obj/item/pipe = 2)
+	tools = list(TOOL_WELDER,TOOL_SAW)
+	time = 150
+	category = CAT_WEAPONRY
+	subcategory = CAT_PARTS
+
+// RECEIVERS
+
+/datum/crafting_recipe/rifle_receiver
+	name = "Improvised Rifle Receiver"
+	result = /obj/item/weaponcrafting/improvised_parts/rifle_receiver
+	reqs = list(/obj/item/stack/sheet/metal = 20)
+	tools = list(TOOL_SCREWDRIVER, TOOL_WELDER) // Rifle is the easiest to craft and can be made at an autolathe, this is a very light kick in the shin for dual-wielding ishotguns.
+	time = 50
+	category = CAT_WEAPONRY
+	subcategory = CAT_PARTS
+
+/datum/crafting_recipe/shotgun_receiver
+	name = "Improvised Shotgun Receiver"
+	result = /obj/item/weaponcrafting/improvised_parts/shotgun_receiver
+	reqs = list(/obj/item/stack/sheet/metal = 10,
+				/obj/item/stack/sheet/plasteel = 1)
+	tools = list(TOOL_SCREWDRIVER, TOOL_WELDER) // Increased cost is to stop dual-wield alpha striking. ishotgun is a rvolver and can be duel-wielded
```

Comment:
Weird spot for this code comment but have you considered locating where ever in gods name the check to stop shotguns from being dual-wielded is and replacing it with a more generic flag?

Reply:
I originally wanted to make them two-handed so they have an actual reasonable position in the game instead of being the the unbalanced as fuck alpha strike for a literal free crit.
Problem is, people will _lose their shit_ if I tried to make the shotgun harder to craft, slightly less damage **and** made them two-handed.
Fortunately, another PR has gone up to make them two-handed, so that should address this.
I'm glad someone else sees the sense in making them two-handed since this PR really doesn't really bring the ishotgun in line as much as it needs to since the alpha strike is still enough to crit.
Once I get a merge conflict for the PR to make ishotguns two-handed I'll sigh in relief.

Reply:
I wanted to make the shotguns two-handed at first, but I was worried that this would get the PR too much backlash. Even though that's the correct balance decision to make (the rifle is twohanded), I'm already nerfing the damage by locking it to inferior ammo and I'm making it more expensive to craft.
Fortunately though, someone else made a PR to make the ishotgun two-handed. Hopefully that gets merged so it'll become two-handed. I suppose once I see a conflict I know that the two-handed PR got merged so I can remove the comment.
I'm glad someone else thinks this should be two-handed though.

Reply:
I don't think anyone is going to yell at you for consistency. It's just I feel like the heavy weapon change fits the bigger shotguns due to using a pump action, while these not is more, to put it in the best terms, 'cinematic'. From a pure gameplay standpoint there is good reason to not have them dual-wieldable. But the fix isn't making them heavy and you can save me and everyone some time having to make that change by making it yourself if you can figure it out.

Ask kev, he may know a solution. Dual-wielding was his code. @kevinz000

Reply:
Just to throw the idea out there, maybe making the full length weapon two-handed, but making the sawn-off one-handed but there's some sort of disadvantage using it without a free-hand or alternatively a disadvantage to sawing it off in the first place could work?
At the moment, there's no reason _not_ to saw it off.

Reply:
Ghommie's PR should solve the two-handed issue I think.